msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -26,25 +27,43 @@
     private static final String TAG = SingleAccountPublicClientApplication.class.getSimpleName();
 
     /**
-     * Name of the shared preference cache for storing current account.
-     * */
+     * Name of the shared preference cache for storing SingleAccountPublicClientApplication data.
+     */
     public static final String SINGLE_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES =
             "com.microsoft.identity.client.single_account_credential_cache";
 
+    /**
+     * SharedPreference key for storing current account.
+     */
+    public static final String CURRENT_ACCOUNT_SHARED_PREFERENCE_KEY =
+            "com.microsoft.identity.client.single_account_credential_cache.current_account";
+
+    private SharedPreferencesFileManager sharedPreferencesFileManager;
+
     protected SingleAccountPublicClientApplication(@NonNull final Context context,
                                                    @NonNull final PublicClientApplicationConfiguration developerConfig) {
         super(context, developerConfig);
+        initializeSharedPreferenceFileManager(context);
     }
 
     protected SingleAccountPublicClientApplication(@NonNull final Context context,
                                                    @NonNull final String clientId) {
         super(context, clientId);
+        initializeSharedPreferenceFileManager(context);
     }
 
     protected SingleAccountPublicClientApplication(@NonNull final Context context,
                                                    @NonNull final String clientId,
                                                    @NonNull final String authority) {
         super(context, clientId, authority);
+        initializeSharedPreferenceFileManager(context);
+    }
+
+    private void initializeSharedPreferenceFileManager(@NonNull final Context context) {
+        sharedPreferencesFileManager = new SharedPreferencesFileManager(
```

Comment:
@shoatman I'll be using this to stored the deserialized MultiTenantAccount. 

Regarding Synchronization, If we want to protect against race condition, we could potentially put the callback function of getCurrentAccount and removeCurrentAccount inside a block. IMHO that's kind of overkill. Please let me know if you think otherwise.

Reply:
Synced offline - one concern here would be an app that calls `removeAccount()` and `getCurrentAccount()` simultaneously.

I would argue that this falls under a usage issue and doesn't warrant a great deal of bulletproofing in the MSAL-side, especially given the _other_ corner-cases we might face regarding `removeAccount()`.... for instance, 1 app calls `removeAccount()` while another app is mid token acquisition... Then we get into a larger discussion about globally synchronizing broker so that `removeAccount()` cannot be called while _any app_ is acquiring tokens. If we're trying to guard against strange race conditions this would seem to be a better angle of attack.

Let's sync in scrum; synchronizing `removeAccount()` in broker such that it can't be invoked while other apps are acquiringTokens might be useful but it would also invite other problems: for instance, an app starts interactive auth and then moves to the background. The lock could be auto-freed after 10 mins, but we can't make anyone wait that long to remove an account realistically.

The more I think about this, the more I think we should fire-off the actions as best-effort and retrofit more robust synchronization if actual, real-life issues arise.